transitgateway.tf

```diff
@@ -26,7 +26,37 @@ resource "aws_ec2_transit_gateway_vpc_attachment" "assessment" {
   vpc_id             = aws_vpc.assessment.id
 }
 
+# Break the association between the transit gateway VPC attachment and
+# the default transit gateway route table.
+#
+# It would be nice not to have to use the Terraform escape hatch for
+# this, but Terraform just doesn't support this sort of cross-account
+# foo right now.
+resource "null_resource" "break_association_with_default_route_table" {
+  # Require that the transit gateway VPC attachment is created before
+  # breaking the association.
+  depends_on = [
+    aws_ec2_transit_gateway_vpc_attachment.assessment,
+  ]
+
+  provisioner "local-exec" {
+    when = create
+    # This command asks AWS to disassociate the default route table
+    # from our transit gateway attachment, then loops until the
+    # disassociation is complete.
+    command = "aws --profile cool-sharedservices-provisionaccount ec2 disassociate-transit-gateway-route-table --transit-gateway-route-table-id ${local.transit_gateway_default_route_table_id} --transit-gateway-attachment-id ${aws_ec2_transit_gateway_vpc_attachment.assessment.id} && while aws --profile cool-sharedservices-provisionaccount ec2 get-transit-gateway-route-table-associations --transit-gateway-route-table-id ${local.transit_gateway_default_route_table_id} | grep --quiet ${aws_vpc.assessment.id}; do sleep 5s; done"
```

Comment:
It's probably possible to:
1. Pull the `cool-sharedservices-provisionaccount` role ID out of the backend state.
2. Run `aws sts` to produce temporary credentials to assume the role.
3. Use the temporary credentials to run the two `aws ec2` commands in the `command` above.

@all - Do you think this is a cleaner approach?  The code will be more complicated, but it will mean that the user need not have `aws-profile-sync` set up locally.

Reply:
I'm 100% on board with your plan to create an issue now and revisit it after we have moved to Terraform 0.13.

Reply:
See issue #74.